Exiled.Events/Features/Event{T}.cs

```diff
@@ -195,14 +195,31 @@ internal void InvokeAsync(T arg)
 
             foreach (CustomAsyncEventHandler<T> handler in InnerAsyncEvent.GetInvocationList().Cast<CustomAsyncEventHandler<T>>())
             {
+                Timing.RunCoroutine(SafeCoroutineEnumerator(handler(arg), handler));
+            }
+        }
+
+        /// <summary>
+        ///     Runs the coroutine manualy so exceptions can be caught and logged.
+        /// </summary>
+        private IEnumerator<float> SafeCoroutineEnumerator(IEnumerator<float> coroutine, CustomAsyncEventHandler<T> handler)
+        {
+            while (true)
+            {
+                object current;
                 try
                 {
-                    Timing.RunCoroutine(handler(arg));
+                    if (!coroutine.MoveNext())
+                        break;
+                    current = coroutine.Current;
                 }
                 catch (Exception ex)
                 {
                     Log.Error($"Method \"{handler.Method.Name}\" of the class \"{handler.Method.ReflectedType.FullName}\" caused an exception when handling the event \"{GetType().FullName}\"\n{ex}");
+                    yield break;
                 }
+
+                yield return (float)current;
```

Comment:
I feel like the Error would come from there and not being catch by the try catch